### ISSUE TITLE
Cyberduck using >170Gb of memory and then crushing to download large folders from a server

### ISSUE THREAD

**Issue by gabersyd** [iterate-ch/cyberduck#15510]
This is a bug. I use a MacStudio, with M1 Ultra processor.

Instead of writing the data on my harddrive, it seems that it is trying to store it in the memory of the computer. 
![cyberduck bug](https://github.com/iterate-ch/cyberduck/assets/20067231/c47559f8-f036-4a67-a55c-a1ad5116e3f4)

**Comment by ylangisc** [comment]
Possible duplicate of #15488. What's the protocol you are using? Any chance to follow the steps in https://github.com/iterate-ch/cyberduck/issues/15488#issuecomment-1876761991 and provide a screenshot when reproducing the issue (memory >2GB)?

**Comment by tristanmcguire** [comment]
This seems to be the same/similar issue to what I have found.  Mac M2 Pro, Sonoma 14.4, downloading via SFTP.  At the beginning of the transfer:  
<img width="1125" alt="Screenshot 2024-04-12 at 16 01 33" src="https://github.com/iterate-ch/cyberduck/assets/43784562/3ff1b678-3d80-4e83-a148-15778fda97d1">

After ~15mins:  
<img width="1123" alt="Screenshot 2024-04-12 at 16 17 28" src="https://github.com/iterate-ch/cyberduck/assets/43784562/01a86832-bb52-4d29-859d-1cc9e0ebbeac">

And after ~30mins:
<img width="1124" alt="Screenshot 2024-04-12 at 16 35 58" src="https://github.com/iterate-ch/cyberduck/assets/43784562/201d8123-6b75-47dd-9d97-f66fe366c11b">


I discovered the issue only after getting the 'out of application memory' error message and things going a bit wonky.  Didn't continue to complete crash, but stopping the transfer, closing and re-opening CyberDuck resets everything and the next download will repeat the process.

**Comment by ylangisc** [comment]
Thanks @tristanmcguire for the screenshots. As I'm not able to reproduce the issue I rely on the reporters to provide the information needed to find the cause. So, it would be great to also get a [VisualVM](https://github.com/iterate-ch/cyberduck/issues/15488#issuecomment-1876761991) screenshot from such a situation.

**Comment by kdambiec** [comment]
I've also encountered this issue quite often on Mac so had to switch from Cyberduck to a different application.

**Comment by ylangisc** [comment]
> I've also encountered this issue quite often on Mac so had to switch from Cyberduck to a different application.

To get a [VisualVM](https://github.com/iterate-ch/cyberduck/issues/15488#issuecomment-1876761991) screenshot would be _very_ helpful to track down the issue as I'm not able to reproduce the issue.

**Comment by whwread** [comment]
Same issue here with DropBox. Mac Mini M2, Sonoma.
Trying to queue large amounts of data to be repatriated.
The Mac Mini will choke itself in the process using all the RAM and about 2x that on the drive as virtual memory until the system runs out of memory. Now using GoodSync, which is 1/3 of the speed. Happy to provide more info if needed.

**Comment by DerekTCS** [comment]
Same issue here, using 38GB on my M1 Pro. Is there a workaround ? I already tried disabling segmented transfers.

**Comment by kdambiec** [comment]
> Same issue here, using 38GB on my M1 Pro. Is there a workaround ? I already tried disabling segmented transfers.

I haven't found any work around, and the developers appear to be unable to reproduce the issue.

**Comment by brinscall** [comment]
45GB of memory used before it crashed. Downloading a large dataset from my Google Drive.
